src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -41,21 +44,23 @@ public void testRestApiRolesEnabled() throws Exception {
             .build();
 
         setup(additionalSettings);
-        TestAuditlogImpl.clear();
-        String body = "{ \"password\":\"some new password\",\"backend_roles\":[\"role1\",\"role2\"] }";
-        HttpResponse response = rh.executePutRequest(
-            "_opendistro/_security/api/internalusers/compuser?pretty",
-            body,
-            encodeBasicHeader("admin", "admin")
-        );
-        Thread.sleep(1500);
-        Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
-        Assert.assertTrue(TestAuditlogImpl.messages.size() + "", TestAuditlogImpl.messages.size() == 1);
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
-        Assert.assertFalse(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
-        validateMsgs(TestAuditlogImpl.messages);
+        Thread.sleep(1000);
+        final List<AuditMessage> messages = TestAuditlogImpl.doThenWaitForMessages(() -> {
```

Comment:
There is `doThenWaitForMessage(...)` which avoids the list which isn't needed since you expect one message, example:

https://github.com/opensearch-project/security/blob/9be1b272f99421cdf7470cf30c44e11f1061f371/src/test/java/org/opensearch/security/auditlog/compliance/ComplianceAuditlogTest.java#L90-L98

Reply:
I will refactor these to use the version that returns a single message

Reply:
done